pkg/sloop/storemanager/storemanager.go

```diff
@@ -191,34 +196,77 @@ func doCleanup(tables typed.Tables, timeLimit time.Duration, sizeLimitBytes int,
 	var totalNumOfDeletedKeys float64 = 0
 	var totalNumOfKeysToDelete float64 = 0
 	anyCleanupPerformed := false
-	if cleanUpTimeCondition(minPartition, maxPartition, timeLimit) || cleanUpFileSizeCondition(stats, sizeLimitBytes) {
+	minPartitionStartTime, err := untyped.GetTimeForPartition(minPartition)
+	if err != nil {
+		return false, 0, 0, err
+	}
+
+	numOfKeysToDeleteForFileSizeCondition := 0.0
+	isFileSizeConditionMet, garbageCollectionRatio := cleanUpFileSizeCondition(stats, sizeLimitBytes, gcThreshold)
+
+	if isFileSizeConditionMet {
+		numOfKeysToDeleteForFileSizeCondition = getNumberOfKeysToDelete(tables.Db(), garbageCollectionRatio)
+	}
+
+	beforeGCTime := time.Now()
+	for cleanUpTimeCondition(minPartition, maxPartition, timeLimit) || numOfKeysToDeleteForFileSizeCondition > 0 {
 		partStart, partEnd, err := untyped.GetTimeRangeForPartition(minPartition)
 		glog.Infof("GC removing partition %q with data from %v to %v (err %v)", minPartition, partStart, partEnd, err)
```

Comment:
consider separating the code to delete a partition in to a new method (separate from the code that determines the need to delete a partition) - this should make the code easier to understand